doc/source/whatsnew/v1.2.0.rst

```diff
@@ -8,6 +8,20 @@ including other versions of pandas.
 
 {{ header }}
 
+.. warning::
+
+   Previously, the default argument ``engine=None`` to ``pd.read_excel``
+   would result in using the xlrd engine in many cases. The engine xlrd is no longer
```

Comment:
double back-tick on xlrd (alt can put a link to xlrd itself, e.g. https://xlrd.readthedocs.io/en/latest/)